metadata-db-app/int-test/cmr/metadata_db/int_test/concepts/utils/association.clj

```diff
@@ -0,0 +1,54 @@
+(ns cmr.metadata-db.int-test.concepts.utils.association
+  "Defines implementations for all of the multi-methods for associations in the metadata-db
+  integration tests."
+  (:require
+   [cmr.metadata-db.int-test.concepts.utils.interface :as concepts]
+   [clojure.string :as string]))
+
+(def association-edn
+  "Valid EDN for generic association metadata"
+  (pr-str {:source-concept-identifier "DQS120000008-PROV1"
+           :source-revision-id 1
+           :associated-concept-id "C120000000-PROV1"
+           :associated-revision-id 1
+           :data {:XYZ "ZYX"}}))
+
+(defmethod concepts/get-sample-metadata :generic-association
+  [_]
+  association-edn)
+
+(defn- create-association-concept
+  "Creates a association concept"
+  [assoc-concept first-concept uniq-num attributes]
+  (let [{:keys [concept-id revision-id]} assoc-concept
+        first-concept-id (:concept-id first-concept)
+        first-revision-id (:revision-id first-concept)
+        user-id (str "user" uniq-num)
+        native-id (string/join "/" [first-concept-id concept-id revision-id])
+        extra-fields (merge {:associated-concept-id concept-id
+                             :associated-revision-id revision-id
+                             :source-concept-identifier first-concept-id
+                             :source-concept-id first-revision-id}
+                            (:extra-fields attributes))
+        attributes (merge {:user-id user-id
+                           :format "application/edn"
+                           :native-id native-id
+                           :extra-fields extra-fields}
+                          (dissoc attributes :extra-fields))]
+    ;; no provider-id should be specified for tool associations
```

Comment:
This seems like there should be internal validation to prevent that from happening, if it can be done accidentally

Reply:
Yes, I have removed the dissoc - it wasn't needed.